File: somephp/dtopattern.php
<?php
class UserDTO {
    public int $id;
    public string $name;
    public string $email;

    public function __construct(int $id, string $name, string $email) {
        $this->id = $id;
        $this->name = $name;
        $this->email = $email;
    }
}

// Użycie
$userDTO = new UserDTO(1, "John Doe", "john@example.com");
var_dump($userDTO);
// object(UserDTO)#1 (3) {
//   ["id"]=>
//   int(1)
//   ["name"]=>
//   string(8) "John Doe"
//   ["email"]=>
//   string(16) "john@example.com"
// }

File: pythonbetter/dtopattern.py
class UserDTO:
    def __init__(self, id: int, name: str, email: str):
        self.id = id
        self.name = name
        self.email = email

# Użycie
user_dto = UserDTO(1, "John Doe", "john@example.com")
print(user_dto.__dict__)
# Output: {'id': 1, 'name': 'John Doe', 'email': 'john@example.com'}